1-Draft/RFCNNNN-Make-Terminating-Errors-Terminate-The-Right-Way.md

```diff
@@ -0,0 +1,176 @@
+---
+RFC: RFCnnnn
+Author: Kirk Munro
+Status: Draft
+SupercededBy: 
+Version: 1.0
+Area: Engine
+Comments Due: July 15, 2019
+Plan to implement: Yes
+---
+
+# Make terminating errors terminate the right way in PowerShell
+
+By default in PowerShell, terminating errors do not actually terminate. For example, if you invoke this command in global scope, you will see the output "Why?" after the terminating error caused by the previous command:
+
+```PowerShell
+& {
+    1/0
+    'Why?'
+}
+```
+
+PowerShell has upheld this behaviour since version 1.0 of the language. You can make the terminating error actually terminate execution of the command, by wrapping the command in try/catch, like this:
+
+```PowerShell
+try {
+    1/0
+    'Why?'
+} catch {
+    throw
+}
```

Comment:
I really hate this - having to use `try` to convert non-terminating errors to terminating just to rethrow the exception.  FWIW, I use this syntax:
```
& {
    $ErrorActionPreference = "Stop"
    "".substring(99)
    "Why?"
}
```

Reply:
I added that example to the RFC. Thanks.